addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1925,11 +1928,13 @@ var SnippetsMenu = Widget.extend({
      * @private
      * @param {OdooEvent} ev
      */
-    _onCloneSnippet: function (ev) {
+    _onCloneSnippet: async function (ev) {
         ev.stopPropagation();
-        this._createSnippetEditor(ev.data.$snippet).then(editor => {
-            editor.clone();
-        });
+        const editor = await this._createSnippetEditor(ev.data.$snippet);
+        await editor.clone();
+        if (typeof ev.data.onSuccess === 'function') {
```

Comment:
I think that just checking `if (ev.data.onSuccess) {` would have sufficied but ok

Reply:
Changed it to be consistent with how it's done elsewhere in the code.